scripts/west_commands/packages.py

```diff
@@ -0,0 +1,168 @@
+            "-m",
+            "--module",
+            action="append",
+            default=[],
+            dest="modules",
+            metavar="<module>",
+            help="Zephyr module to run the 'packages' command for. "
+            "Use 'zephyr' if the 'packages' command should run for Zephyr itself. "
+            "Option can be passed multiple times. "
+            "If this option is not given, the 'packages' command will run for Zephyr "
+            "and all modules.",
+        )
+
+        subparsers_gen = parser.add_subparsers(
+            metavar="<manager>",
+            dest="manager",
+            help="select a manager.",
+            required=True,
+        )
+
+        pip_parser = subparsers_gen.add_parser(
+            "pip",
+            help="manage pip packages",
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            epilog=textwrap.dedent(
+                """
+            Manage pip packages:
+
+                Run 'west packages pip' to print all pip packages and requirements
+                files needed by Zephyr and modules.
+
+                The output is compatible with the requirements file format itself.
+            """
+            ),
+        )
+
+        pip_parser.add_argument(
+            "--install",
+            action="store_true",
+            help="Install pip packages instead of listing them. "
+            "A single 'pip install' command is built and executed. "
+            "Additional pip arguments can be passed after a -- separator "
+            "from the original 'west packages pip --install' command. For example pass "
+            "'--dry-run' to pip not to actually install anything, but print what would be.",
+        )
+
+        return parser
+
+    def do_run(self, args, unknown):
+        if len(unknown) > 0 and unknown[0] != "--":
+            self.die(
+                f'Unknown argument "{unknown[0]}"; '
+                'arguments for the manager should be passed after "--"'
+            )
+
+        # Store the zephyr modules for easier access
+        self.zephyr_modules = zephyr_module.parse_modules(ZEPHYR_BASE, self.manifest)
+
+        if args.modules:
+            # Check for unknown module names
+            module_names = [m.meta.get("name") for m in self.zephyr_modules]
+            module_names.append("zephyr")
+            for m in args.modules:
+                if m not in module_names:
+                    self.die(f'Unknown zephyr module "{m}"')
+
+        if args.manager == "pip":
+            return self.do_run_pip(args, unknown[1:])
+
+        # Unreachable but print an error message if an implementation is missing.
+        self.die(f'Unsupported package manager: "{args.manager}"')
+
+    def do_run_pip(self, args, manager_args):
+        packages = []
+        requirements = []
+
+        if not args.modules or "zephyr" in args.modules:
+            requirements.append(ZEPHYR_BASE / "scripts/requirements.txt")
```

Comment:
Note: We also have `doc/requirements.txt`.

Reply:
These are not installed as per the getting started guide.

Reply:
I know, just wanted to point it out for completeness.